src/clausetable.jl

```diff
@@ -0,0 +1,115 @@
+
+"Nested dictionary to store indexed clauses."
+const ClauseTable = Dict{Symbol,Dict{Symbol,Vector{<:AbstractClause}}}
+
+function insert_clause!(table::ClauseTable, c::Clause)
```

Comment:
The idea was indeed to define a different method for different subtypes of `AbstractClause`. I think this makes it easier to extend. This way I can keep everything related to Problog in a separate package, instead of modifying Julog with annotated disjunctions which are not really necessary for pure-Prolog functionality.